### PR TITLE
core: add boot-loop detection

### LINKED ISSUE #2639: blueos_startup_update should have a mechanism to prevent boot loops

### PR DESCRIPTION
**Pull request by Williangalvani**
simple fix for #2639

It can be tested with williangalvani/blueos-core:fox_boot_loop_test

which includes this patch and the broken commit from #2649

fix #2639

### CODE REVIEW

File: core/start-blueos-core [hunk 1]
```diff
@@ -41,6 +43,8 @@ echo "Total RAM size: ${TOTAL_RAM_MB} MB"
 if [ -z "${BLUEOS_DISABLE_STARTUP_UPDATE}" ]; then
     blueos_startup_update.py || true  # We NEVER want to prevent booting
 fi
+# if we made it here, it means the patcher didn't require a reboot
+rm $BOOT_LOOP_DETECTOR
```
Comment (Williangalvani) [inline core/start-blueos-core:47]: this actually could be inside blueos_startup_update.py

File: core/tools/blueos_startup_update/blueos_startup_update.py [hunk 2]
```diff
@@ -448,6 +450,10 @@ def run_command_is_working():
 
 def main() -> int:
     start = time.time()
+    # check if boot_loop_detector exists
+    if os.path.isfile(BOOT_LOOP_DETECTOR):
+        logger.warning("It seems we are in a boot loop, skipping patches.")
```
Comment (patrickelectric) [inline core/tools/blueos_startup_update/blueos_startup_update.py:455]: "It seems that we are starting from a previous startup configuration, skipping patches."

Comment (Williangalvani) [reply core/tools/blueos_startup_update/blueos_startup_update.py:455]: that sounds... vague
what about "It seems we the startup patches were just applied on the last boot. skipping patches..." ?

Comment (patrickelectric) [reply core/tools/blueos_startup_update/blueos_startup_update.py:455]: Yep

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**